site/components/DateSelector.tsx

```diff
@@ -1,81 +1,100 @@
-                        type="text"
-                        defaultValue={inputs?.dayInput}
-                    />
-                </div>
-            </div>
-
-            <div className="govuk-date-input__item">
-                <div className="govuk-form-group">
-                    <label
-                        className="govuk-label govuk-date-input__label"
-                        htmlFor={type ? `${type}-${startOrEnd}-month-input` : `${startOrEnd}-month-input`}
-                    >
-                        Month
-                    </label>
-                    <input
-                        className={`govuk-input govuk-date-input__input govuk-input--width-2 ${
-                            errors.length > 0 ? "govuk-input--error" : ""
-                        } `}
-                        id={`${type}-${startOrEnd}-month-input`}
-                        name={`${type}${type ? startCase(startOrEnd) : startOrEnd}DateMonth`}
-                        type="text"
-                        defaultValue={inputs?.monthInput}
-                    />
-                </div>
-            </div>
-            <div className="govuk-date-input__item">
-                <div className="govuk-form-group">
-                    <label
-                        className="govuk-label govuk-date-input__label"
-                        htmlFor={type ? `${type}-${startOrEnd}-year-input` : `${startOrEnd}-year-input`}
-                    >
-                        Year
-                    </label>
-                    <input
-                        className={`govuk-input govuk-date-input__input govuk-input--width-4 ${
-                            errors.length > 0 ? "govuk-input--error" : ""
-                        } `}
-                        id={`${type}-${startOrEnd}-year-input`}
-                        name={`${type}${type ? startCase(startOrEnd) : startOrEnd}DateYear`}
-                        type="text"
-                        defaultValue={inputs?.yearInput}
-                    />
-                </div>
+const inputBox = (
+    startOrEnd: "start" | "end",
+    hasErrors: boolean,
+    inputRef: React.Ref<any> | undefined,
+    inputProps: InputBaseComponentProps | undefined,
+    InputProps: Partial<FilledInputProps> | Partial<OutlinedInputProps> | undefined,
+    disabled: boolean,
+    inputId: string,
+) => (
+    <div className="govuk-date-input flex items-center [&_.MuiSvgIcon-root]:fill-govBlue" id={`${startOrEnd}-date`}>
+        <div className="govuk-date-input__item govuk-!-margin-right-0">
+            <div className="govuk-form-group">
+                <input
+                    className={`govuk-input govuk-date-input__input govuk-input--width-6 ${
+                        hasErrors ? "govuk-input--error" : ""
+                    } `}
+                    id={inputId}
+                    name={`${inputId.includes("publish") ? "publish" : "validity"}-${startOrEnd}DateDay`}
```

Comment:
I'd be tempted to pass this through as a prop as well, then this component doesn't need to care about publish or validity and can leave it up to the parent, also I'd remove the hyphen from the name since I think we generally use camelcase for the input names

Reply:
Yeah sure